developer/src/tike/project/Keyman.Developer.UI.Project.UpgradeProject.pas

```diff
@@ -29,22 +34,32 @@ function TryUpgradeProject(Project: TProject): TUpgradeResult;
     Exit;
   end;
 
-{  if not FKeymanDeveloperOptions.PromptForProjectUpgrade then
+{ TODO:
+ if not FKeymanDeveloperOptions.PromptForProjectUpgrade then
   begin
     // User wishes to stick with v1.0 projects
     Exit;
-  end;}
+  end;
+}
+
+  frmMessages.Clear;
 
   if not Project.CanUpgrade then
   begin
     // Project has restrictions, such as files in wrong folders, so
     // we cannot upgrade. Show a message for the user
-    ShowMessage('The current project cannot be upgraded to v2.0. The following errors were encountered:'#13#10+
-      Project.UpgradeMessages.Text);
+    for msg in Project.UpgradeMessages do
+    begin
+      Project.Log(plsError, Project.FileName, msg, 0, 0);
+    end;
+    MessageDlg('Some issues must be addressed before the current project can '+
+      'be upgraded to Keyman Developer 17 format.  '+
```

Comment:
l.61 calls it "Keyman Developer 17.0 format".
Do we future-proof with Keyman Developer 17.0+ ....?

Reply:
I thought about it but I think it's a Developer 17 file format and 18 may have a new format. I'm okay with keeping it as is :grin: